core/templates/services/common-events.service.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service to mimic the functionality of $rootScope.
+ */
+
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CommonEventsService {
+  // TODO(srijanreddy98): Remove the _rootScope once the project is migrated.
+  private _rootScope;
+  private _loadingMessage = new Subject<string>();
```

Comment:
One concern here is why we are using `Subject` and not `Observable`?

Reply:
Observables by default are cold. Instead of making them hot (so that they can be broadcast to multiple subscribers), I have just used subjects which means a lot less code. As far as I know, one can't push new values to an observable (outside of where you create it). Basically a subject is a hot observable with the ability to pump in new values.

My views are based off this medium article (https://medium.com/@benlesh/on-the-subject-of-subjects-in-rxjs-2b08b7198b93). It is written by Ben Lesh, who is an RxJS lead.

Reply:
I didn't get the word "hot" and "cold". 

> As far as I know, one can't push new values to an observable (outside of where you create it). Basically a subject is a hot observable with the ability to pump in new values.

Could you explain this may be with the help of an example?

Reply:
Writing about hot and cold observables could be a bit long, I would refer this to an excellent read from the makers of RxJS
https://medium.com/@benlesh/hot-vs-cold-observables-f8094ed53339